### PR TITLE
Switch to geth types

### PR DESCRIPTION
**Pull request by lightclient**

### CODE REVIEW

File: api/builder.go [hunk 1]
```diff
@@ -2,30 +2,47 @@ package api
...
+	GasUsed          uint64         `json:"gasUsed"       gencodec:"required"`
+	Timestamp        uint64         `json:"timestamp"     gencodec:"required"`
+	ExtraData        []byte         `json:"extraData"     gencodec:"required"`
+	BaseFeePerGas    *big.Int       `json:"baseFeePerGas" gencodec:"required"`
+	BlockHash        common.Hash    `json:"blockHash"     gencodec:"required"`
+	TransactionsRoot common.Hash    `json:"transactionsRoot"  gencodec:"required"`
+}
+
+// JSON type overrides for executableData.
+type executionPayloadHeaderMarshalling struct {
```
Comment (protolambda) [inline api/builder.go:38]: I think these type overrides are a horrible hack, but it's also nice to have `uint64` in the interface ofc., so it's ok-ish

File: api/engine.go [hunk 2]
```diff
@@ -181,29 +106,40 @@ const (
 	ExecutionInvalidTerminalBlock ExecutePayloadStatus = "INVALID_TERMINAL_BLOCK"
 )
 
+// PayloadID is an identifier of the payload build process
+type PayloadID [8]byte
```
Comment (protolambda) [inline api/engine.go:110]: Instead you can import `beacon.PayloadID`. See https://github.com/ethereum/go-ethereum/blob/f0328f241b7c3def217b0c2dce1a1b297f979a37/core/beacon/types.go#L88 (the nice payload ID type is upstream now :) )

**Review by protolambda — COMMENTED** [review]
Looks good to me. Unfortunate we need `gencodec`, but it works. Would import `PayloadID` from geth instead.

File: api/engine.go (hunk 2 above)
Comment (lightclient) [reply api/engine.go]: Done! Thanks

File: api/builder.go (hunk 1 above)
Comment (lightclient) [reply api/builder.go:38]: yeah :(

**Pull request merged**